docs/docs/sample-scripts/entra/find-obsolete-m365-groups/index.mdx

```diff
@@ -0,0 +1,293 @@
+    param (
+      [Parameter(Mandatory = $false)] [Switch] $KeepOutputPath
+    )
+
+    if (!$KeepOutputPath.IsPresent -or $null -eq $Global:Path) {
+      $Script:Path = Read-Host "Set the path to the folder where you want to export the report data as csv file"
+    }
+    
+    $TestPath = Test-Path -Path  $Script:Path
+    $tStamp = (Get-Date).ToString("yyyy-MM-dd HH:mm:ss")
+    if ($TestPath -ne $true) {
+      New-Item -ItemType directory -Path $Script:Path | Out-Null
+      Write-Host "Will create file in $($Script:Path): M365GroupsReport-{current date}.csv" -ForegroundColor Yellow
+    }
+    else {
+      Write-Host "Following report file will be created in $($Script:Path): 'M365GroupsReport-$($tStamp).csv'."
+      Write-Host "`nAll data will be exported to $($Script:Path): M365GroupsReport-$($tStamp).csv." -ForegroundColor Blue
+      Write-Host "Do not edit this file during the scan." -ForegroundColor Blue
+    }
+    $Script:ReportPath = "$($Script:Path)/M365GroupsReport-$($tStamp).csv"
+  }
+
+  function Get-AllM365Groups {
+    $groups = m365 entra m365group list --includeSiteUrl | ConvertFrom-Json
+    $Script:Groups = $groups | Where-Object { $null -ne $_.siteUrl }
+  }
+
+  function Start-GroupInsightsTests {
+    Write-Host "Checking $($Script:Groups.Count) groups for activity"
+
+    $Script:Groups | ForEach-Object {
+      $groupInfo = [GroupInfo]::new()
+      $groupInfo.Reference = $_
+      $groupInfo.Membership = @{Owners = 0; Members = 0; Guests = 0 }
+      $groupInfo.TestStatus = "🟢 OK"
+
+      Write-Host "☀︎ $($groupInfo.Reference.displayName)"
+
+      # Tests
+      Test-GroupMembership -Group $groupInfo
+      Test-SharePointActivity -Group $groupInfo
+      Test-ConversationActivity -Group $groupInfo
+
+      # Report
+      New-Report -Group $groupInfo
+    }
+    
+    #Give feedback to user
+    Write-Host "`n-------------------------------------------------------------------"
+    Write-Host "`SUMMARY" -ForegroundColor DarkGreen
+    Write-Host "`-------------------------------------------------------------------"
+    Write-Host "`n👉 Found $($Global:ObsoleteGroups.Count) group$($Global:ObsoleteGroups.Count -gt 1 ? 's' : '') with possibly low activity."
+    Write-Host "`   Please review the report: " -NoNewline
+    Write-Host "$($Script:ReportPath)" -ForegroundColor DarkBlue
+  }
+
+  function Test-GroupMembership {
+    [CmdletBinding()]
+    param (
+      [Parameter(Mandatory = $true)] [GroupInfo] $Group
+    )
+
+    # Original lists
+    $users = m365 entra m365group user list --groupId $Group.Reference.id | ConvertFrom-Json
+    $owners = m365 entra m365group user list --groupId $Group.Reference.id --role Owner | ConvertFrom-Json
+    $members = m365 entra m365group user list --groupId $Group.Reference.id --role Member | ConvertFrom-Json
+    
+    # Consider guests as users that are not in the $members list
+    $guests = Compare-Object -ReferenceObject $users $members -PassThru
```

Comment:
Ok! Thanks for pointing that out, @milanholemans. I didn't know about this behaviour when having 'everyone m365 groups' in the tenant. (At least for me) this is a special behavior that I would not have expected – I suspect that this is due to the fact that 'everyone' points directly to an 'all users' group in Entra ID, since the maximum number of users per group is limited. I will look into this though.

But: shouldn't we also include in the documentation (https://pnp.github.io/cli-microsoft365/cmd/entra/group/group-user-list/) that if group is an Viva engage 'all company group', the return values for `Owner` and `Member` are empty arrays?

Reply:
Tbh I don't think it's worth mentioning it. We also don't want to clutter the docs with too many remarks.